### ISSUE TITLE
-H:Log=registerResource only works when combined with --no-server

### ISSUE THREAD

**Issue by abelsromero** [oracle/graal#1652]
Contrary to what documentation https://github.com/oracle/graal/blob/master/substratevm/RESOURCES.md#L43 says, exact name inclusion (example: `.*/Resource0.txt$`) does not work.

In this example https://github.com/abelsromero/graal-resource-reader, you can test that `-H:IncludeResources='.*/config.properties$'` does not include the file, but `-H:IncludeResources='.*/*.properties$'` does.

As a side note: I don't see any output with `-H:Log=registerResource` or `-H:Log=registerResource:verbose`.

Graal version: 19.2.0
OS: Linux 5.1.21-1-MANJARO

**Comment by olpaw** [comment]
Hi @abelsromero ,

unfortunately for `-H:Log=registerResource` to show something on stdout you also have to pass `--no-server`. Sorry for that! We will fix the issue.`

When using `--no-server -H:Log=registerResource -H:IncludeResources='.*/*.properties$'` the log output shows what's going on.
```
...
  [thread:26] scope: ForkJoinPool-4-worker-1.registerResource
  ResourcesFeature: registerResource: config.properties
  [thread:26] scope: ForkJoinPool-4-worker-1.registerResource
  ResourcesFeature: registerResource: META-INF/maven/org.abelsromero.native/graal-resource-reader/pom.properties
...
```
As you can see from the log output, a resource in the root of a classpath directory does **not** start with `/`. That's also why `-H:IncludeResources='.*/config.properties$'` does not work. If you use 
```
-H:IncludeResources='^config.properties$'
```
instead it works as expected.

As a side note `-H:IncludeResources='.*/*.properties$'` merely works by accident. Let's see:
* `.*` ... any char zero or more times
* `/*` ... followed by a `/`-char zero or more times (zero times in case of our accidental match)
* `.*` ... followed by any char zero or more times
* `properties$` ... followed by char-sequence `properties` followed by a newline.

**Comment by abelsromero** [comment]
I must apologize, I have to admit I suspected it had something to do with `*./` but it was late. Thanks to you great explanation now I see my mistake reading the regex. I also tested with `-H:IncludeResources='config.properties'` and it works :+1: 

btw, feel free to change the issue title to what you think better reflect the issue at hand. I understand the bug tag is about `-H:Log=registerResource`.
